drivers/storage/libstorage/libstorage_client_xcli.go

```diff
@@ -19,56 +19,57 @@ import (
 
 func (c *client) Supported(
 	ctx types.Context,
-	opts types.Store) (bool, error) {
+	opts types.Store) (types.LSXSupportedOp, error) {
 
 	if c.isController() {
-		return false, utils.NewUnsupportedForClientTypeError(
+		return 0, utils.NewUnsupportedForClientTypeError(
 			c.clientType, "Supported")
 	}
 
 	ctx = context.RequireTX(ctx.Join(c.ctx))
 
 	serviceName, ok := context.ServiceName(ctx)
 	if !ok {
-		return false, goof.New("missing service name")
+		return 0, goof.New("missing service name")
 	}
 
 	si, err := c.getServiceInfo(serviceName)
 	if err != nil {
-		return false, err
+		return 0, err
 	}
 	driverName := strings.ToLower(si.Driver.Name)
 
 	// check to see if the driver's executor is supported on this host
 	if ok := c.supportedCache.IsSet(driverName); ok {
-		return c.supportedCache.GetBool(driverName), nil
+		return c.supportedCache.GetLSXSupported(driverName), nil
 	}
 
 	out, err := c.runExecutor(ctx, driverName, types.LSXCmdSupported)
 	if err != nil {
 		if err == types.ErrNotImplemented {
 			ctx.WithField("serviceDriver", driverName).Warn(
 				"supported cmd not implemented")
-			c.supportedCache.Set(driverName, true)
+			c.supportedCache.Set(driverName, types.LSXOpAllNoMount)
```

Comment:
I realize that this code here isn't new, just modified, but I'm surprised it's hear. I thought the containing function here was only called if we had already done a type assertion that the executor implemented the interface that includes the `Supported()` method. As such, I would consider it an actual error if it's defined in the executor, but returns a Not Implemented error. I think this is probably here for backwards compatibility, but is that an issue at this point?

I guess I'm really wondering if it is safe to assume that if `Supported` is not implemented, we say this is `LSXOpAllNomount`, and I believe the answer to that is yes, since that's what we require from a minimal Executor. Which probably means this whole comment is moot. :)